### PR TITLE
Allow accessing external environments in Delta Lake query runner

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

The main purpose is quickly accessing product test environments using the new `DeltaLakeExternalQueryRunnerMain`.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java [hunk 1]
```diff
@@ -263,20 +263,18 @@ public static void main(String[] args)
...
-    public static class DeltaLakeGlueQueryRunnerMain
+    public static class DeltaLakeExternalQueryRunnerMain
     {
         public static void main(String[] args)
                 throws Exception
         {
-            // Requires AWS credentials, which can be provided any way supported by the DefaultProviderChain
-            // See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
-            DistributedQueryRunner queryRunner = builder(createSession())
+            // Please set Delta Lake connector properties via VM options. e.g. -Dhive.metastore.uri=thrift://localhost:9083 -D..
```
Comment (krvikash) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java:271]: the comment says to set `hive.metastore.uri`. But for running Delta query runner with Glue metastore requires setting `hive.metastore`.

How about if we can have `DeltaLakeGlueQueryRunnerMain` and `DeltaLakeExternalQueryRunnerMain` both?

Comment (krvikash) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java:271]: Can we update the comment then to include `hive.metastore`?

**Review by krvikash — APPROVED** [review]

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java [hunk 2]
```diff
@@ -125,10 +119,18 @@ public DistributedQueryRunner build()
     public static DistributedQueryRunner createDeltaLakeQueryRunner(String catalogName, Map<String, String> extraProperties, Map<String, String> connectorProperties)
             throws Exception
     {
+        Path metastoreDirectory = Files.createTempDirectory(catalogName);
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java:122]: We should create the path only inside the `if` statement.

**Review by findinpath — APPROVED** [review]

**Comment by findepi** [comment]
> This allows accessing product test environment with VM options.

I use `DevelopmentServer` for that
https://github.com/trinodb/trino/blob/022a266cf6ac8eea18ac4319a0591c2e41d9352c/testing/trino-server-dev/etc/catalog/hive.properties#L10-L11

**Review by findepi — APPROVED** [review]
i wouldn't want this (see conversation), but whatever.

**Pull request merged**